### PR TITLE
Refactor: convert main navbar main components to hooks

### PR DESCRIPTION
**Pull request by martiuh**
Also add a `useStyles` hook

### CODE REVIEW

File: src/components/NavBar/NavBar.jsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React, { useState, useEffect } from "react";
 import { NavBarRightSide } from "./NavBarRightSide";
 import { NavBarWrapper } from "./NavBarWrapper";
 import { AvatarButton } from "./AvatarButton";
-import { UserNameLabel } from "./UsernameLabel";
+import { UsernameLabel } from "./UsernameLabel";
```
Comment (mishelashala) [inline src/components/NavBar/NavBar.jsx]: refactor to:

```tsx
import { UserNameLabel } from "./UserNameLabel";
```

File: src/components/NavBar/NavBarLeftSide.tsx [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import React from "react";
 import Grid from "@material-ui/core/Grid";
 
-export const NavBarLeftSide = ({ children }) => {
+export interface INavBarLeftSideProps {
+  children: React.ReactChildren;
```
Comment (mishelashala) [inline src/components/NavBar/NavBarLeftSide.tsx]: no need for `children` on props, by default react/ts adds it. If the interface its empty then just remove it

File: src/components/NavBar/NavBarRightSide.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import Grid from "@material-ui/core/Grid";
+
+export interface INavBarRightSide {
+  children: React.ReactChildren;
```
Comment (mishelashala) [inline src/components/NavBar/NavBarRightSide.tsx]: same here, no need for children/interface

File: src/components/NavBar/NavBarWrapper.tsx [hunk 4]
```diff
@@ -1,19 +1,30 @@
...
   toolBar: {
     backgroundColor: "#3049a1",
-    zIndex: theme.zIndex.drawer + 1
+    zIndex: 10
   }
 });
 
-const NavBarWrapperComponent = ({ children, classes: { appBar, toolBar } }) => {
+export interface INavBarWrapperComponentProps {
+  children: React.ReactChildren;
```
Comment (mishelashala) [inline src/components/NavBar/NavBarWrapper.tsx]: no children

File: src/components/NavBar/UsernameLabel.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { Typography } from "@material-ui/core/";
+
```
Comment (mishelashala) [inline src/components/NavBar/UsernameLabel.tsx]: delete blank space

**Review by mishelashala — CHANGES_REQUESTED** [review]

File: src/components/NavBar/NavBarLeftSide.tsx (hunk 2 above)
Comment (martiuh) [reply src/components/NavBar/NavBarLeftSide.tsx]: I get an error when I try to start/build the app
```bash
Binding element 'children' implicitly has an 'any' type.  TS7031
```

**Review by mishelashala — APPROVED** [review]

**Pull request merged**